### PR TITLE
Update build

### PR DESCRIPTION
**Pull request by SuperQ**
* Update Makefile.
* Update travis config.
* Fix staticcheck errors.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,8 +3,10 @@ sudo: false
 language: go
 
 go:
-- 1.7.5
-- 1.8
+- 1.8.x
```
Comment (mdlayher) [inline .travis.yml]: Now that 1.9 is out, do you want to just do `1.x` or also add `1.9.x`?

File: tcp_test.go [hunk 2]
```diff
@@ -35,7 +35,7 @@ func TestTCPConnection(t *testing.T) {
 	go func() {
 		conn, err := ln.Accept()
 		if err != nil {
-			t.Fatalf("Error accepting on socket: %s", err)
+			panic(fmt.Sprintf("Error accepting on socket: %s", err))
```
Comment (mdlayher) [inline tcp_test.go]: I sometimes make a `panicf` helper for these cases, but it isn't super necessary for just a handful.

Comment (brian-brazil) [inline tcp_test.go]: Why are these changing? They look correct to me.

File: Makefile [hunk 3]
```diff
@@ -52,7 +59,16 @@ docker:
 	@echo ">> building docker image"
 	@docker build -t "$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TAG)" .
 
-$(PROMU) promu:
+$(GOPATH)/bin/promu promu:
 	@GOOS= GOARCH= $(GO) get -u github.com/prometheus/promu
 
-.PHONY: all style format build test vet tarball docker promu
+$(GOPATH)/bin/staticcheck:
```
Comment (brian-brazil) [inline Makefile]: This is not in the standard makefile, please keep in sync with upstream

File: tcp_test.go (hunk 2 above)
Comment (SuperQ) [reply tcp_test.go]:
```
the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (SA2002)
```

File: Makefile (hunk 3 above)
Comment (SuperQ) [reply Makefile]: [staticcheck](https://staticcheck.io/) is used in several other exporters.

Comment (brian-brazil) [reply Makefile]: It's not in the promu or prometheus makefiles, I don't want exporters to get their makefiles out of sync.

File: Makefile [hunk 4]
```diff
@@ -11,18 +11,21 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-GO     := GO15VENDOREXPERIMENT=1 go
+GO     ?= GO15VENDOREXPERIMENT=1 go
 GOPATH := $(firstword $(subst :, ,$(shell $(GO) env GOPATH)))
-PROMU  ?= $(GOPATH)/bin/promu
-pkgs    = $(shell $(GO) list ./... | grep -v /vendor/)
+GOARCH := $(shell $(GO) env GOARCH)
```
Comment (brian-brazil) [inline Makefile]: There's FIRST_GOPATH stuff in promu, we should keep this all in sync

**Comment by SuperQ** [comment]
@brian-brazil Ok, I've synced up the Makefile with https://github.com/prometheus/prometheus/pull/3125.

**Comment by brian-brazil** [comment]
CI is failing, looks like the Makefile from promu isn't all there.

**Comment by SuperQ** [comment]
Yea, working on it.

File: .travis.yml (hunk 1 above)
Comment (SuperQ) [reply .travis.yml]: Looks like we still need 1.9 in our tools.

**Pull request merged**